compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -95,21 +97,16 @@ loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
     }
     else
     {
-      // We use shape from the node itself
-      loco::TensorShape shape;
-      shape.rank(node->rank());
-      for (uint32_t r = 0; r < node->rank(); ++r)
+      auto shape_node = loco::must_cast<luci::CircleNode *>(node->shape());
+      assert(shape_node->rank() == 1);
+      // shape_node tensor values will provide new shape, like [2, 3, 4]
+      auto num_elements = shape_node->dim(0).value(); // above example will give 3
+      shape_by_input.rank(num_elements);
+      for (uint32_t r = 0; r < shape_by_input.rank(); ++r)
       {
-        // TODO remove this copy from `use_own(node);`
-        // Shape inference rules in this file did not consider unknown dimension.
-        // If some node has unknown dimension, 0 is inserted and wrong shape
-        // inference was done as a result.
-        // To fix this, new shape inference algorithm is being implemented.
-        // Until new inference algorithm is fully implemented, unknown dimension
-        // would be represented as 1 along with TFLite expression.
-        shape.dim(r) = node->dim(r).known() ? node->dim(r).value() : 1;
+        shape_by_input.dim(r).unset();
       }
-      shape_by_input = shape;
+      is_static_shape = false;
```

Comment:
I think `for(...)` loop with `shape_by_input.dim(r).unset();` can be removed.
There was another PR, by SAFFY, that used like this.
`shape_by_input.rank(num_elements);` itself will set all dims to unknown(same as unset)

Reply:
Thanks for your information :)
I'll make the change.